pkg/usersettings/handlers.go

```diff
@@ -0,0 +1,263 @@
+	}
+
+	_, err = client.RbacV1().RoleBindings(USER_SETTINGS_NAMESPACE).Create(context, roleBinding, meta.CreateOptions{})
+	if err != nil {
+		klog.Warningf("Could not create user-settings Role \"%s\" for user \"%s\" (UID: %s)", resourceData.getRoleBindingName(), resourceData.Username, resourceData.UID)
+	}
+
+	configMap, err = client.CoreV1().ConfigMaps(USER_SETTINGS_NAMESPACE).Create(context, configMap, meta.CreateOptions{})
+	if err != nil {
+		klog.Warningf("Could not create user-settings ConfigMap \"%s\" for user \"%s\" (UID: %s)", resourceData.getConfigMapName(), resourceData.Username, resourceData.UID)
+		// Return actual ConfigMap if it is already created
+		if apierrors.IsAlreadyExists(err) {
+			return client.CoreV1().ConfigMaps(USER_SETTINGS_NAMESPACE).Get(context, resourceData.getConfigMapName(), meta.GetOptions{})
+		} else {
+			return nil, err
+		}
+	} else {
+		return configMap, nil
+	}
+}
+
+// Fetch the current user with auth.User token and deletes corresponding user-setting resources.
+// It handles not found responses, so that it does not fail on multiple calls.
+func (h *UserSettingsHandler) deleteUserSettings(user *auth.User) error {
+	context := context.TODO()
+
+	resourceData, err := h.getResourceData(context, user)
+	if err != nil {
+		return err
+	}
+
+	client, err := h.createServiceAccountClient()
+	if err != nil {
+		return err
+	}
+
+	err = client.RbacV1().RoleBindings(USER_SETTINGS_NAMESPACE).Delete(context, resourceData.getRoleBindingName(), meta.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	err = client.RbacV1().Roles(USER_SETTINGS_NAMESPACE).Delete(context, resourceData.getRoleName(), meta.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	err = client.CoreV1().ConfigMaps(USER_SETTINGS_NAMESPACE).Delete(context, resourceData.getConfigMapName(), meta.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	return nil
+}
+
+func (h *UserSettingsHandler) createServiceAccountClient() (*kubernetes.Clientset, error) {
+	config := &rest.Config{
+		Host:        h.Endpoint,
+		BearerToken: h.ServiceAccountToken,
+		Transport:   h.Client.Transport,
+	}
+	return kubernetes.NewForConfig(config)
+}
+
+func (h *UserSettingsHandler) createUserProxyClient(user *auth.User) (dynamic.Interface, error) {
+	config := &rest.Config{
+		Host:        h.Endpoint,
+		BearerToken: user.Token,
+		Transport:   h.Client.Transport,
+	}
+	return dynamic.NewForConfig(config)
+}
+
+func (h *UserSettingsHandler) getResourceData(context context.Context, user *auth.User) (*ResourceData, error) {
+	client, err := h.createUserProxyClient(user)
+	if err != nil {
+		return nil, err
+	}
+
+	userInfo, err := client.Resource(USER_RESOURCE).Get(context, "~", meta.GetOptions{})
+	if err != nil {
+		klog.Infof("getCurrentUser error: %v", err)
+		return nil, err
+	}
+
+	name := string(userInfo.GetName())
```

Comment:
`GetName()` is returning a string.
```suggestion
	name := string(userInfo.GetName())
```